### PR TITLE
executor: execute parallel unsafe ops serially

### PR DESCRIPTION
**Pull request by abn**
Resolves: #2658 #3086
Closes: #3095

### CODE REVIEW

File: poetry/installation/executor.py [hunk 1]
```diff
@@ -107,14 +107,30 @@ def execute(self, operations):  # type: (Operation) -> int
...
+                    serial_operations.append(operation)
+                    continue
+
                 tasks.append(self._executor.submit(self._execute_operation, operation))
 
             try:
                 wait(tasks)
+
+                for operation in serial_operations:
+                    wait([self._executor.submit(self._execute_operation, operation)])
```
Comment (PetterS) [inline poetry/installation/executor.py:132]: Don't have to submit and wait here. Can just call `_execute_operation`.

**Review by PetterS — COMMENTED** [review]
Thanks a lot for this! 👍 

I think that in order to test this, some refactoring of the code is needed.

File: poetry/installation/executor.py (hunk 1 above)
Comment (abn) [reply poetry/installation/executor.py:135]: Yup, but I wanted keep calls consistent via `self._executor` for now.

**Comment by abn** [comment]
> I think that in order to test this, some refactoring of the code is needed.

Race conditions are inherently hard to test. I would be happy with no regressions and manual verification for now.

**Review by finswimmer — APPROVED** [review]
LGTM :+1:

**Pull request merged**